### ISSUE TITLE
new feature vmagent: Influx line protocol but on socket TCP/UDP

### ISSUE THREAD

**Issue by syepes** [VictoriaMetrics/VictoriaMetrics#333]
**Is your feature request related to a problem? Please describe.**
Migrate off Telegraf :-)

**Describe the solution you'd like**
Implement a socket listener for the Influx line protocol

**Describe alternatives you've considered**
For the moment the only viable solution is to use Telegraf, If the vmagent can directly receive and buffer these metric we can remove the middleman

**Additional context**
Below is an example of the socket listener

https://github.com/influxdata/telegraf/tree/master/plugins/inputs/socket_listener/
telegraf.conf (Example)
```
[[inputs.socket_listener]]
  service_address = "udp://:8188"
  data_format = "influx"

[[inputs.socket_listener]]
  service_address = "tcp://:8189"
  data_format = "influx"
```

**Comment by valyala** [comment]
I'm going to implement this via `-influxListenAddr` command-line flag similar to `-graphiteListenAddr` and `-opentsdbListenAddr`. If this flag is set, then vmagent will accept Influx line protocol data over tcp and udp defined in the flag.

**Comment by valyala** [comment]
The commit 6386aeb1e004b896e2c1de134ee63b04a0edb66e adds an ability to accept Influx line protocol data to `vmagent` and VictoriaMetrics via TCP and UDP if `-influxListenAddr` command-line flag is set. This commit will be included into the next release. In the meantime the functionality can be tested via building `vmagent` from sources according to [these docs](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/app/vmagent/README.md#how-to-build-from-sources).

**Comment by syepes** [comment]
Great addition, thanks!

**Comment by valyala** [comment]
This feature is available starting from [v1.34.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).
